src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -21,61 +43,689 @@ const ActorInit Door_Shutter_InitVars = {
+            this->unk_170 = 1.0f;
+            return true;
+        }
+        DoorShutter_SetupAction(this, func_80996F98);
+        return false;
+    } else if (doorType == SHUTTER_BACK_LOCKED) {
+        DoorShutter_SetupAction(this, func_80996B00);
+        return false;
+    }
+    DoorShutter_SetupAction(this, func_80996B0C);
+    return false;
+}
+
+void DoorShutter_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DoorShutter* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
+    s32 phi_a3;
+    s32 pad;
+    s32 objectIndex;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.initPosRot.pos.z = this->dyna.actor.shape.unk_08;
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK);
+    this->doorType = (this->dyna.actor.params >> 6) & 0xF;
+    phi_a3 = D_80998224[this->doorType];
+    if (phi_a3 < 0) {
+        ShutterSceneInfo* phi_v1;
+
+        for (phi_v1 = &D_80998240[0], i = 0; i < ARRAY_COUNT(D_80998240) - 1; i++, phi_v1++) {
+            if (globalCtx2->sceneNum == phi_v1->sceneNum) {
+                break;
+            }
+        }
+        phi_a3 = phi_v1->index;
+    } else if (phi_a3 == 6) {
+        BossDoorInfo* phi_v1_2;
+
+        for (phi_v1_2 = &D_80998288[0], i = 0; i < ARRAY_COUNT(D_80998288) - 1; i++, phi_v1_2++) {
+            if (globalCtx2->sceneNum == phi_v1_2->dungeonScene || globalCtx2->sceneNum == phi_v1_2->bossScene) {
+                break;
+            }
+        }
+        this->unk_168 = phi_v1_2->index;
+    } else {
+        this->dyna.actor.room = -1;
+    }
+    if (this->requiredObject = objectIndex = Object_GetIndex(&globalCtx2->objectCtx, D_809980F0[phi_a3].objectId),
+        (s8)objectIndex < 0) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    DoorShutter_SetupAction(this, DoorShutter_SetupType);
+    this->unk_16B = phi_a3;
+    if (this->doorType == SHUTTER_KEY_LOCKED || this->doorType == SHUTTER_BOSS) {
+        if (!Flags_GetSwitch(globalCtx2, this->dyna.actor.params & 0x3F)) {
+            this->unk_16E = 10;
+        }
+        Actor_SetHeight(&this->dyna.actor, 60.0f);
+    } else if (phi_a3 == 4) {
+        Actor_SetScale(&this->dyna.actor, 0.1f);
+        this->unk_166 = 100;
+        this->dyna.actor.uncullZoneScale = 200.0f;
+        Actor_SetHeight(&this->dyna.actor, 0.0f);
+    } else {
+        Actor_SetHeight(&this->dyna.actor, 60.0f);
+    }
+}
+
+void DoorShutter_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DoorShutter* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    if (this->dyna.actor.room >= 0) {
+        s32 transitionActorId = (u16)this->dyna.actor.params >> 0xA;
+
+        globalCtx->transitionActorList[transitionActorId].id *= -1;
+    }
+}
+
+void DoorShutter_SetupType(DoorShutter* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->requiredObject)) {
```

Comment:
Same suggestion as for En_Door regarding `requiredObject`
```suggestion
    if (Object_IsLoaded(&globalCtx->objectCtx, this->requiredObjBankIndex)) {
```